Sources/Defaults/DefaultsBridge.swift

```diff
@@ -0,0 +1,185 @@
+import Foundation
+
+public protocol DefaultsBridge {
+	// The type of Value of Key<Value>
+	associatedtype Value
+
+	// This type should be one of the NativelySupportedType
+	associatedtype Serializable
+
+	// Serialize Value to Serializable before we store it in UserDefaults
+	func serialize(_ value: Value?) -> Serializable?
+
+	// Deserialize Serializable to Value
+	func deserialize(_ object: Serializable?) -> Value?
+}
+
+public struct DefaultsURLBridge: DefaultsBridge {
+	public func serialize(_ value: URL?) -> Any? {
+		if let value = value {
+			if #available(macOS 10.13, watchOS 4.0, macOSApplicationExtension 10.13, watchOSApplicationExtension 4.0, *) {
```

Comment:
This is not correct. The way to encode URL should not depend on the platform version. That was not how the old logic worked.